### PR TITLE
Code Updates with few changes

### PR DESCRIPTION
**Pull request by kshitijbhat330**
Fillna not working in labels. Please look into it

### CODE REVIEW

File: Code/CheXpertDataset.py [hunk 1]
```diff
@@ -43,28 +48,41 @@ def __len__(self):
...
 
-        return im, label
+        return im, torch.from_numpy(np.array(label[label_columns].values.tolist()))
 
 
 class ReplaceNaNTransform:
     def __call__(self, sample):
-        return sample[label_columns].fillna(-2)
-
+        sample[label_columns].fillna(-2)
```
Comment (josh-bernstein618) [inline Code/CheXpertDataset.py:67]: I don't see how this change would work.  According to the documentation, `fillna` is by default non-desctructive

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.fillna.html

File: Code/Data_Download.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Please execute this script only once. The script download several gigabytes of data from AWS S3 bucket
+import os
+
+os.system("sh ../download_kaggle_data.sh")
+os.sytem("sh ../download_stanford_data.sh")
```
Comment (josh-bernstein618) [inline Code/Data_Download.py:5]:
```suggestion
os.system("sh ../download_stanford_data.sh")
```

File: Code/training_common_utils.py [hunk 3]
```diff
@@ -1,25 +1,27 @@
...
     transforms.Resize((600, 600)),
     transforms.ToTensor(),
-    transforms.Normalize((0.0, 0.0, 0.0), (1.0, 1.0, 1.0))
+    transforms.Normalize((0, 0, 0), (1.0, 1.0, 1.0))
 ])
 
 label_preprocessing_uzeros = transforms.Compose([
     UZerosTransform(),
-    ReplaceNaNTransform()
+    ReplaceNaNTransform(),
```
Comment (josh-bernstein618) [inline Code/training_common_utils.py:19]: I personally am not a fan of hanging commas, even when the language allows it

File: download_kaggle_data.sh [hunk 4]
```diff
@@ -6,3 +6,26 @@ echo "Download is proceeding in the background."
...
+then
+  mkdir Data
+  if [ $? -ne 0]:
+  then
+    echo " Error creating the directory. Check your permissions"
+    exit(1)
+  fi
+fi
+
+unzip CheXpert-v1.0-small.zip -d Data/
```
Comment (josh-bernstein618) [inline download_kaggle_data.sh:20]: +1

File: Code/training_common_utils.py (hunk 3 above)
Comment (zbuckley) [reply Code/training_common_utils.py:19]:
```suggestion
    ReplaceNaNTransform()
```

File: Code/training_common_utils.py [hunk 5]
```diff
@@ -163,6 +168,7 @@ def training_loop(
         with torch.no_grad():
             loss_validation = []
             for (images, labels) in validation_loader:
+                images = images.squeeze(0)
```
Comment (zbuckley) [inline Code/training_common_utils.py:171]: strange.. never had to do this previously. though i may not gotten the additional dimension somehow...

**Review by zbuckley — APPROVED** [review]

File: Code/training_common_utils.py [hunk 6]
```diff
@@ -1,25 +1,27 @@
...
 label_preprocessing_uzeros = transforms.Compose([
     UZerosTransform(),
-    ReplaceNaNTransform()
+    ReplaceNaNTransform(),
 ])
 
 label_preprocessing_uones = transforms.Compose([
     UOnesTransform(),
-    ReplaceNaNTransform()
+    ReplaceNaNTransform(),
```
Comment (josh-bernstein618) [inline Code/training_common_utils.py:24]: Just so we don't forget
```suggestion
    ReplaceNaNTransform()
```

File: Code/CheXpertDataset.py (hunk 1 above)
Comment (zbuckley) [reply Code/CheXpertDataset.py:67]: looks like the only thing changed was the return being removed so the behaviour hasn't changed right? 
fillna being the right answer here could be a good question to ask, but i don't think it's worth holding up the pr.

Comment (zbuckley) [reply Code/CheXpertDataset.py:67]: oh.. i see the return below.

Comment (zbuckley) [reply Code/CheXpertDataset.py:67]: based on the docs we need to do this: 
```suggestion
        sample[label_columns].fillna(-2, inplace=True)
```

File: Code/training_common_utils.py [hunk 7]
```diff
@@ -140,7 +143,9 @@ def training_loop(
 
         model.train()
         # assuming loader will return image, label tuple
+        print("The type of training_loader is:",type(training_loader))
         for i, (images, labels) in enumerate(training_loader):
+            images = images.squeeze(0)
```
Comment (zbuckley) [inline Code/training_common_utils.py:148]: same thought as below... but I'm willing to believe it.

File: Code/training_team8.py [hunk 8]
```diff
@@ -47,6 +51,7 @@
 
 # %% -------------------------------------- Training Prep ----------------------------------------------------------
 model = densenet121(num_classes=14).to(device)
+model.avgpool = nn.AdaptiveAvgPool2d(1)
```
Comment (zbuckley) [inline Code/training_team8.py:54]: sure thing if it helps, but i don't understand it atm. Comment/links to help explain should be included.

**Review by zbuckley — CHANGES_REQUESTED** [review]
See various discussions on-going.
